errai-common/src/main/java/org/jboss/errai/common/client/dom/elemental2/Elemental2DomUtil.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.errai.common.client.dom.elemental2;
+
+import com.google.gwt.user.client.ui.RootPanel;
+import com.google.gwt.user.client.ui.Widget;
+import elemental2.dom.HTMLElement;
+import elemental2.dom.Node;
+import jsinterop.base.Js;
+
+/**
+ * Provides utility methods for interacting with the DOM.
+ * @author Guilherme Carreiro <ggomes@redhat.com>
+ */
+public class Elemental2DomUtil {
+
+    /**
+     * Detaches all element children from a node.
+     * @param node Must not be null.
+     * @return True iff any element children were detached by this call.
+     */
+    public static boolean removeAllElementChildren(final Node node) {
+
+        final boolean hadChildren = node.lastChild != null;
```

Comment:
Maybe could be more defensive: `final boolean hadChildren = node != null && node.lastChild != null;`, not sure let me know what do you think.

Reply:
I don't think it's a good idea to silently return if a null is passed in. We could enhance it to assert the parameter is non-null, but it's not a big deal for me since a null would cause an NPE on this line where it would be obvious that node was the culprit.

Reply:
@mbarkley thanks for you comment, it sounds reasonable to me, just to explain my point a little bit more, I wanted to prevent NPE because it is public util method, which doesn't mention this NPE behavior in javadoc, that is why I thought NPE shouldn't be thrown so easy, @karreiro leaving the decision on you, if you prefer variant with NPE, could we have mentioned this in javadoc?

Reply:
Thanks for the comment @jomarko.

However, I agree with @mbarkley, a NullPointerException on this line is clear enough.

I prefer to keep it as is, to keep it simple (or simpler).

Reply:
FWIW the javadoc looks good to me, because it already states that the node parameter must not be null. Saying that a parameter must not be null makes it fairly obvious what the problem was when you get an NPE.